R/checks.R

```diff
@@ -33,12 +33,16 @@ check_grid <- function(grid, workflow, pset = NULL) {
   }
 
   if (nrow(pset) == 0L) {
-    msg <- paste0(
-      "No tuning parameters have been detected, ",
-      "performance will be evaluated using the resamples with no tuning. ",
-      "Did you want to [tune()] parameters?"
+    msg <- c(
+      "!" = "No tuning parameters have been detected;
+             performance will be evaluated using the resamples with no tuning."
     )
-    rlang::warn(msg)
+
+    if (nrow(pset) == nrow(tune_args(workflow))) {
```

Comment:
```suggestion
    if (nrow(tune_args(workflow)) == 0L) {
```

Maybe slightly clearer since you are already in the `nrow(pset) == 0L` branch